File: templates/Uwp/Features/FirstUsePrompt.Prism/Services/FirstRunDisplayService.cs
using System;
using System.Threading.Tasks;
using Microsoft.Toolkit.Uwp.Helpers;
using Windows.ApplicationModel.Core;
using Windows.UI.Core;
using Param_RootNamespace.Views;

namespace Param_RootNamespace.Services
{
    public class FirstRunDisplayService : IFirstRunDisplayService
    {
        private static bool shown = false;

        public async Task ShowIfAppropriateAsync()
        {
            await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(
                CoreDispatcherPriority.Normal, async () =>
                {
                    if (SystemInformation.IsFirstRun && !shown)
                    {
                        shown = true;
                        var dialog = new FirstRunDialog();
                        await dialog.ShowAsync();
                    }
                });
        }
    }
}

File: templates/Uwp/Features/FirstUsePrompt._VB/Services/FirstRunDisplayService.vb
Imports Param_RootNamespace.Views

Imports Microsoft.Toolkit.Uwp.Helpers

Imports Windows.ApplicationModel.Core
Imports Windows.UI.Core

Namespace Services
    Public NotInheritable Class FirstRunDisplayService
        Shared shown As Boolean = False

        Private Sub New()
        End Sub

        Friend Shared Async Function ShowIfAppropriateAsync() As Task
            Await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal,
            Async Sub()
                If SystemInformation.IsFirstRun AndAlso Not shown Then
                    shown = True
                    Dim dialog = New FirstRunDialog()
                    Await dialog.ShowAsync()
                End If
            End Sub)
        End Function
    End Class
End Namespace
